src/dodal/devices/oav/snapshots/snapshot_with_beam_centre.py

```diff
@@ -0,0 +1,63 @@
+from ophyd_async.core import SignalR
+from PIL import Image, ImageDraw
+
+from dodal.devices.areadetector.plugins.MJPG import MJPG
+
+
+class SnapshotWithBeamCentre(MJPG):
+    """A child of MJPG which, when triggered, draws an outlined crosshair at the beam
+    centre in the image and saves the image to disk."""
+
+    CROSSHAIR_LENGTH_PX = 20
+    CROSSHAIR_OUTLINE_COLOUR = "Black"
+    CROSSHAIR_FILL_COLOUR = "White"
+
+    def __init__(
+        self,
+        prefix: str,
+        beam_x_signal: SignalR,
+        beam_y_signal: SignalR,
+        name: str = "",
+    ) -> None:
+        self.beam_centre_i = beam_x_signal
+        self.beam_centre_j = beam_y_signal
+        super().__init__(prefix, name)
+
+    async def post_processing(self, image: Image.Image):
+        beam_x = await self.beam_centre_i.get_value()
+        beam_y = await self.beam_centre_j.get_value()
+        SnapshotWithBeamCentre.draw_crosshair(image, beam_x, beam_y)
+
+        await self._save_image(image)
+
+    @classmethod
```

Comment:
Should: I know that you didn't do this but I dislike class methods. If it is independent of the class then I think just put it in a separate function